src/Services/QueuedJob.php

```diff
@@ -84,6 +84,23 @@ public function prepareForRestart();
      */
     public function getJobType();
 
+    /**
+     * Specifies what user ID should be hwen runnning the job
+     * valid values:
+     * null - (default) - run the job as current user
+     * 0 - run the job without a user
+     * greater than zero - run the job as a specifi user
+     *
+     * thisis useful in situation like:
+     * - a job needs to always run without user (like static cache job)
+     * - a job needs to run as a specific user (for example data migration job)
+     *
+     * Note that this value can be overriden in the @see QueuedJobService::queueJob()
+     *
+     * @return int|null
+     */
+    public function getRunAsMember();
```

Comment:
Thoughts on calling this `getRunAsMemberID()` since it returns an ID rather than a Member instance?